### ISSUE TITLE
[HELP] Games not visible from Raspberry Pi4 server (games on USB HDD)

### ISSUE THREAD

**Issue by bladerunner94** [shawly/docker-ps3netsrv#27]
**I have read and followed [all troubleshooting steps from the README.md](https://github.com/shawly/docker-ps3netsrv#troubleshooting)?**

```
// SSBoYXZlIHJlYWQsIGZvbGxvd2VkIGFuZCB0cmllZCBldmVyeSB0cm91Ymxlc2hvb3Rpbmcgc3RlcCwgYnV0IGl0IHN0aWxsIGRvZXNuJ3Qgd29yaywgcGxlYXNlIGhlbHAgbWUu
```

**System configuration:**  
pi@SRV:~ $ docker run -d \
    --name=ps3netsrv \
    -p 38008:38008 \
    -v /mnt/usbstorage/shared/ps3games:/games:rw \
    -e USER_ID=1000 \
    -e GROUP_ID=1000 \
    shawly/ps3netsrv:edge
Unable to find image 'shawly/ps3netsrv:edge' locally
edge: Pulling from shawly/ps3netsrv
9b18e9b68314: Already exists
3e264053f6bb: Pull complete
5340e95e1a13: Pull complete
3d0cfd521542: Pull complete
28e2e20a671a: Pull complete
Digest: sha256:76ec0a0bad691edd8926f582f9a32a28fe897b168612976fe0af1a23760eecec
Status: Downloaded newer image for shawly/ps3netsrv:edge
13705a706a153112d49d25412e2e57a8d21871e48f930eccfd2adac5d8d973fe
pi@SRV:~ $ ls -l /mnt/usbstorage/shared/ps3games
total 56
drwxrwxrwx 2 pi pi  4096 Jul  7  2020 BDISO
drwxrwxrwx 2 pi pi  4096 Sep 17  2021 DOWNLOADS
drwxrwxrwx 2 pi pi  4096 Jul  7  2020 DVDISO
drwxrwxrwx 2 pi pi 12288 Jul  7  2020 exdata
drwxrwxrwx 3 pi pi  4096 Aug 22 14:20 GAMES
drwxrwxrwx 6 pi pi 12288 Jul 30 20:06 PKG
drwxrwxrwx 2 pi pi  4096 Jul  7  2020 PS2ISO
drwxrwxrwx 2 pi pi  4096 Aug 22 13:56 PS3ISO
drwxrwxrwx 2 pi pi  4096 Jul  7  2020 PSPISO
drwxrwxrwx 2 pi pi  4096 Jul  7  2020 PSXISO
pi@SRV:~ $ docker exec ps3netsrv ls -l /games                                   total 56
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Jul  7  2020 BDISO
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Sep 17  2021 DOWNLOADS
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Jul  7  2020 DVDISO
drwxrwxrwx 3 ps3netsrv ps3netsrv  4096 Aug 22 15:20 GAMES
drwxrwxrwx 6 ps3netsrv ps3netsrv 12288 Jul 30 21:06 PKG
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Jul  7  2020 PS2ISO
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Aug 22 14:56 PS3ISO
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Jul  7  2020 PSPISO
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Jul  7  2020 PSXISO
drwxrwxrwx 2 ps3netsrv ps3netsrv 12288 Jul  7  2020 exdata
pi@SRV:~ $

Output of `docker version`:
```
// pi@SRV:~ $ docker exec ps3netsrv ls -l /games                                   total 56
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Jul  7  2020 BDISO
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Sep 17  2021 DOWNLOADS
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Jul  7  2020 DVDISO
drwxrwxrwx 3 ps3netsrv ps3netsrv  4096 Aug 22 15:20 GAMES
drwxrwxrwx 6 ps3netsrv ps3netsrv 12288 Jul 30 21:06 PKG
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Jul  7  2020 PS2ISO
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Aug 22 14:56 PS3ISO
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Jul  7  2020 PSPISO
drwxrwxrwx 2 ps3netsrv ps3netsrv  4096 Jul  7  2020 PSXISO
drwxrwxrwx 2 ps3netsrv ps3netsrv 12288 Jul  7  2020 exdata
```

Output of `docker ps -a -f "ancestor=shawly/ps3netsrv"`:
```
// docker ps -a -f "ancestor=shawly/ps3netsrv:edge"
CONTAINER ID   IMAGE                   COMMAND   CREATED         STATUS         PORTS                                           NAMES
6b51932134b6   shawly/ps3netsrv:edge   "/init"   5 minutes ago   Up 5 minutes   0.0.0.0:38008->38008/tcp, :::38008->38008/tcp   ps3netsrv

```

Output of `docker logs CONTAINERNAME` (replace *CONTAINERNAME* with your ps3netsrv container's name):
```
// 
pi@SRV:~ $ docker logs ps3netsrv
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-adduser: executing...
usermod: no changes

-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

[cont-init.d] 00-adduser: exited 0.
[cont-init.d] done.
[services.d] starting services
2022-10-03 20:55:16.448336868  ps3netsrv build 20220813 (mod by aldostools)
[services.d] done.

```

Output of `id` (you should be logged in with the user that owns your games directory):
```
// 
pi@SRV:~ $ id pi
uid=1000(pi) gid=1000(pi) groups=1000(pi),4(adm),20(dialout),24(cdrom),27(sudo),29(audio),44(video),46(plugdev),60(games),100(users),104(input),106(render),108(netdev),999(spi),998(i2c),997(gpio),117(lpadmin),994(docker)

```

Output of `docker exec --user ps3netsrv CONTAINERNAME id` (replace *CONTAINERNAME* with your ps3netsrv container's name):
```
// 
pi@SRV:~ $ docker exec --user ps3netsrv ps3netsrv id
uid=1000(ps3netsrv) gid=1000(ps3netsrv) groups=1000(ps3netsrv),100(users)

```

Output of `ls -l /path/to/your/games/folder`:
```
// paste the output of the ls -l /path/to/your/games/folder command here
```

Output of `docker exec CONTAINERNAME ls -l /games` (replace *CONTAINERNAME* with your ps3netsrv container's name):
```
// pi@SRV:~ $ ls -l /mnt/usbstorage/shared/ps3games
total 56
drwxrwxrwx 2 pi pi  4096 Jul  7  2020 BDISO
drwxrwxrwx 2 pi pi  4096 Sep 17  2021 DOWNLOADS
drwxrwxrwx 2 pi pi  4096 Jul  7  2020 DVDISO
drwxrwxrwx 2 pi pi 12288 Jul  7  2020 exdata
drwxrwxrwx 3 pi pi  4096 Aug 22 14:20 GAMES
drwxrwxrwx 6 pi pi 12288 Jul 30 20:06 PKG
drwxrwxrwx 2 pi pi  4096 Jul  7  2020 PS2ISO
drwxrwxrwx 2 pi pi  4096 Aug 22 13:56 PS3ISO
drwxrwxrwx 2 pi pi  4096 Jul  7  2020 PSPISO
drwxrwxrwx 2 pi pi  4096 Jul  7  2020 PSXISO
```

My docker run command or my docker-compose.yml:
```
// docker run -d \
    --name=ps3netsrv \
    -p 38008:38008 \
    -v /mnt/usbstorage/shared/ps3games:/games:rw \
    -e USER_ID=1000 \
    -e GROUP_ID=1000 \
    shawly/ps3netsrv:edge
```

Hello, none of the games (ISO's) are found by webman mod on my ps3. It was working with normal ps3netsrv install on my pi4. Now, with the docker version (after I stopped the ps3netsrv installed on the pi 4), I am not able to see any of my PS3 ISO's. I tried edge and latest builds of ps3netsrv docker image.

**Comment by shawly** [comment]
I updated the image last weekend, can you try using `shawly/ps3netsrv:v1.6.0` instead of edge or latest? 
Maybe the latest build is bugged or not compatible with your installed webMAN-MOD version, you are using the latest webMAN-MOD version right?

Does `docker exec ps3netsrv ls -l /games/GAMES` or `docker exec ps3netsrv ls -l /games/PS3ISO` show all of your games?

Also which standalone version did you use on your Pi4 before using Docker?

**Comment by bladerunner94** [comment]
Thank you, I try your new image and report back. Upon running your 2 suggested commands, I got listed all my ISO's and my single (folder) beta game of NFS MW 2012:

pi@SRV:~ $ docker exec ps3netsrv ls -l /games/GAMES
total 4
drwxrwxrwx 15 ps3netsrv users 4096 Jul  7  2020 HAWAII_MAIN
pi@SRV:~ $ docker exec ps3netsrv ls -l /games/PS3ISO
total 347013336
-rwxrwxrwx 1 ps3netsrv users      6033047552 Nov 14  2018 Alone In The Dark Infe                                                                                                                                   rno-[BLUS30232].iso
-rwxrwxrwx 1 ps3netsrv users      2369030144 Nov 26  2019 Ape Escape On The Move                                                                                                                                   -[BCAS20159].iso
-rwxrwxrwx 1 ps3netsrv users      7182942208 Nov 13  2018 BATTLEFIELD Bad Compan                                                                                                                                   y-[BLES00261].iso
-rwxrwxrwx 1 ps3netsrv ps3netsrv  3378511872 Feb  3  2020 BURNOUT.ISO
-rwxrwxrwx 1 ps3netsrv users      6444920832 Nov 17  2019 Blur-[BLES00759].iso
-rwxrwxrwx 1 ps3netsrv users      7184121856 Oct 19  2019 DIRT 3 Complete Editio                                                                                                                                   n-[BLES01548].iso
-rwxrwxrwx 1 ps3netsrv users      3678994432 Nov 11  2019 F.E.A.R.-[BLUS30003].i                                                                                                                                   so
-rwxrwxrwx 1 ps3netsrv users     12158763008 Dec  8  2018 Formula One Championsh                                                                                                                                   ip Edition-[BCES00005].iso
-rwxrwxrwx 1 ps3netsrv users     13357678592 Nov 15  2019 Genji Days of the Blad                                                                                                                                   e-[BCES-00002].iso
-rwxrwxrwx 1 ps3netsrv users     17754357760 Dec  8  2018 Gran Turismo 5-[BCES00                                                                                                                                   569].iso
-rwxrwxrwx 1 ps3netsrv users     24158863360 Nov 16  2019 Killzone 3-[BCUS98234]                                                                                                                                   .iso
-rwxrwxrwx 1 ps3netsrv users     18619570176 Nov 23  2019 LittleBigPlanet 2 Extr                                                                                                                                   as Edition (EU)-[BCES01693].iso
-rwxrwxrwx 1 ps3netsrv ps3netsrv 20170043392 Jul 31 13:42 MOTORSTORM APOCALYPSE                                                                                                                                    [BCES00484].iso
-rwxrwxrwx 1 ps3netsrv users      3671654400 Dec  8  2015 MOTORSTORM PACIFIC RIF                                                                                                                                   T DEMO [BCES00129].iso
-rwxrwxrwx 1 ps3netsrv ps3netsrv 12883546112 Jul 31 13:17 MOTORSTORM PACIFIC RIF                                                                                                                                   T [BCES00129].iso
-rwxrwxrwx 1 ps3netsrv users      4840947712 Nov 12  2019 Medal of Honor Airborn                                                                                                                                   e-[BLES00174].iso
-rwxrwxrwx 1 ps3netsrv ps3netsrv 17231161344 Jul 31 13:29 MotorStorm [BCES00006]                                                                                                                                   .iso
-rwxrwxrwx 1 ps3netsrv ps3netsrv  5170284544 Aug 22 14:28 NBA 07 (USA).iso
-rwxrwxrwx 1 ps3netsrv users     12162301952 Nov 15  2018 Need For Speed Carbon-                                                                                                                                   [BLUS30016].iso
-rwxrwxrwx 1 ps3netsrv users      7983333376 Nov 14  2018 Need For Speed Prostre                                                                                                                                   et-[BLES00176].iso
-rwxrwxrwx 1 ps3netsrv users      6945439744 Dec  8  2018 Need For Speed Underco                                                                                                                                   ver-[BLES00450].iso
-rwxrwxrwx 1 ps3netsrv ps3netsrv 19914883072 Sep 16  2021 PS3VOLUME.iso
-rwxrwxrwx 1 ps3netsrv users     16802971648 Dec  8  2018 Race Driver GRID-[BLES                                                                                                                                   00256].iso
-rwxrwxrwx 1 ps3netsrv users     16543711232 Nov 11  2019 Resistance Fall of Man                                                                                                                                   -[BCUS98107].iso
-rwxrwxrwx 1 ps3netsrv users     11957239808 Nov 11  2019 SIREN Blood Curse-[BCE                                                                                                                                   S00294].iso
-rwxrwxrwx 1 ps3netsrv users     18544758784 Nov 16  2019 SOCOM 4 US Navy SEALs-                                                                                                                                   [BCUS98135].iso
-rwxrwxrwx 1 ps3netsrv users      2499084288 Nov 24  2019 Start the Party-[BCES0                                                                                                                                   0969].iso
-rwxrwxrwx 1 ps3netsrv users     13504946176 Nov 23  2019 THE FIGHT-[BCES00874].                                                                                                                                   iso
-rwxrwxrwx 1 ps3netsrv users     17518573568 Nov 16  2019 TIME CRISIS RAZING STO                                                                                                                                   RM-[BLUS30528].iso
-rwxrwxrwx 1 ps3netsrv users      9372829696 Nov 16  2019 Test Drive Unlimited 2                                                                                                                                   -[BLES00884].iso
-rwxrwxrwx 1 ps3netsrv users      5676007424 Nov 15  2018 The Club-[BLES00192].i                                                                                                                                   so
-rwxrwxrwx 1 ps3netsrv users      6011027456 Dec  8  2018 The Elder Scrolls IV O                                                                                                                                   blivion GOTY-[BLUS30087].iso
-rwxrwxrwx 1 ps3netsrv users      1257111552 Aug 29  2019 The Shoot-[BCES00463].                                                                                                                                   iso
-rwxrwxrwx 1 ps3netsrv users      2358837248 Nov 14  2019 Way of the Samurai 3-[                                                                                                                                   BLUS30398].iso
pi@SRV:~ $

**Comment by bladerunner94** [comment]
My webman version is the latest: v 1.72.42 (FULL). With "legacy" ps3netsrv, from the same PI, everything was working.

Unfortunately same simptom with v1.6.0 install of your container:

pi@SRV:~ $ docker run -d \
    --name=ps3netsrv \
    -p 38008:38008 \
    -v /mnt/usbstorage/shared/ps3games:/games:rw \
    -e USER_ID=1000 \
    -e GROUP_ID=1000 \
    shawly/ps3netsrv:v1.6.0
Unable to find image 'shawly/ps3netsrv:v1.6.0' locally
v1.6.0: Pulling from shawly/ps3netsrv
810d54c7e01b: Pull complete
77843bd9458f: Pull complete
8f2a5c14184a: Pull complete
8489436fbc33: Pull complete
e0136056349b: Pull complete
Digest: sha256:d6a926bb0c004b2e1c557b3ceb9fc74be5845e00a2a13c3f10db65c3a853cb81
Status: Downloaded newer image for shawly/ps3netsrv:v1.6.0
f9a3ce7bc72d3955483f9003a74ac230f5437784ac6afaf75b650520a5bf8cb4
pi@SRV:~ $

**Comment by bladerunner94** [comment]
Before docker, I used: ps3netsrv_20220421, I complied it myself for the pi4.

**Comment by bladerunner94** [comment]
I am happy to confirm: your edge and v1.5.2 build as well working now successfully, I was able to scan all my games! The issue was, that once I disabled the legacy ps3netsrv service on my PI4, the webman setting: "Scan for LAN games/videos PS3NETSRV#1 IP:..." got un-tick. I ticked back, and tested both your v1.5.2 and edge versions too. This untick "bug" in webman happens after I remove the container, scan for games, 0 results. Then I manually need to tick the box again and scan my contents again.I used the edge version now, tested with NFS Carbon and Resistance: Fall of Man, which are (almost) fully disc based game, without game data installation. Did not encountered any performance issues. 

Is there any way I can contribute to proof that your container works on Raspberry PI4 with a HEN PS3 super slim? Thank you for the help!
